hledger-lib/Hledger/Read.hs

```diff
@@ -126,8 +127,11 @@ defaultJournalPath = do
 --
 readJournalFiles :: Maybe StorageFormat -> Maybe FilePath -> Bool -> [PrefixedFilePath] -> IO (Either String Journal)
 readJournalFiles mformat mrulesfile assrt prefixedfiles = do
-  (either Left (Right . mconcat) . sequence)
+  (right mconcat1 . sequence)
```

Comment:
Why can't we use `mconcat` here?

As I understand the only difference between `mconcat1` and `mconcat` is that custom implementation avoids doing `mappend (last x) mempty` and starts with `mappend (last (init x)) (last x)`.
If there is some issues with that maybe we should re-consider type `Journal`.